// @vercel/sdk/esm/funcs/apiBillingGetV1BillingCharges.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod/v3";
import * as M from "../lib/matchers.js";
import { compactMap } from "../lib/primitives.js";
import { pathToFunc } from "../lib/url.js";
import { APIPromise } from "../types/async.js";
/**
 * Get FOCUS v1.3 compliant usage cost metrics
 *
 * @remarks
 * Returns usage charges in FOCUS v1.3 JSONL format
 */
export function apiBillingGetV1BillingCharges(client, options) {
    return new APIPromise($do(client, options));
}
async function $do(client, options) {
    const path = pathToFunc("/v1/billing/charges")();
    const headers = new Headers(compactMap({
        Accept: "*/*",
    }));
    const context = {
        options: client._options,
        baseURL: options?.serverURL ?? client._baseURL ?? "",
        operationID: "get_/v1/billing/charges",
        oAuth2Scopes: null,
        resolvedSecurity: null,
        securitySource: null,
        retryConfig: options?.retries
            || client._options.retryConfig
            || { strategy: "none" },
        retryCodes: options?.retryCodes || ["429", "500", "502", "503", "504"],
    };
    const requestRes = client._createRequest(context, {
        method: "GET",
        baseURL: options?.serverURL,
        path: path,
        headers: headers,
        userAgent: client._options.userAgent,
        timeoutMs: options?.timeoutMs || client._options.timeoutMs || -1,
    }, options);
    if (!requestRes.ok) {
        return [requestRes, { status: "invalid" }];
    }
    const req = requestRes.value;
    const doResult = await client._do(req, {
        context,
        errorCodes: ["400", "401", "403", "404", "4XX", "500", "503", "5XX"],
        retryConfig: context.retryConfig,
        retryCodes: context.retryCodes,
    });
    if (!doResult.ok) {
        return [doResult, { status: "request-error", request: req }];
    }
    const response = doResult.value;
    const [result] = await M.match(M.nil("2XX", z.void()), M.fail([400, 401, 403, 404, "4XX"]), M.fail([500, 503, "5XX"]))(response, req);
    if (!result.ok) {
        return [result, { status: "complete", request: req, response }];
    }
    return [result, { status: "complete", request: req, response }];
}
